File: hsg9-tinh/2023-2024-haiduong/phuongtrinh/phuongtrinh.py
import os
import sys

input_file = os.path.join(sys.path[0], 'phuongtrinh.inp')
output_file = os.path.join(sys.path[0], 'phuongtrinh.out')

n = 0
count_x = 0


def input():
    global n

    with open(input_file, 'r') as f:
        n = int(f.readline())


def sum_digits(a):
    Sum = 0
    while a > 0:
        Sum += a % 10
        a = a // 10
    
    return Sum


def process():
    global n, count_x

    # Khởi tạo giá trị xuất phát của nghiệm x
    start_x = max(1, n - 81)

    # Duyệt từng giá trị x từ start_x đến n 
    for x in range(start_x, n):
        if x + sum_digits(x) == n:
            count_x += 1


def output():
    global count_x

    with open(output_file, 'w') as f:
        f.write(str(count_x))


if __name__ == '__main__':
    input()
    process()
    output()

File: hsg9-tinh/2023-2024-haiduong/phuongtrinh/phuongtrinh.cpp
#include <iostream>

#define input_file "phuongtrinh.inp"
#define output_file "phuongtrinh.out"

using namespace std;

int n;
int count_x = 0;


void input()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    freopen(input_file, "r", stdin);

    cin >> n;
}


int sum_digits(int a)
{
    int Sum = 0;
    while (a > 0)
    {
        Sum += a % 10;
        a /= 10;
    }

    return Sum;
}


void process()
{
    // Khởi tạo giá trị xuất phát của nghiệm x
    int start_x = max(1, n - 81);

    // Duyệt từng giá trị x từ start_x đến n
    for (int x = start_x; x < n; ++x)
    {
        if (x + sum_digits(x) == n)
        {
            count_x++;
        }
    }
}


void output()
{
    freopen(output_file, "w", stdout);

    cout << count_x;
}


int main()
{
    input();
    process();
    output();

    return 0;
}